build.sh

```diff
@@ -2,6 +2,33 @@
 
 set -e -u -o pipefail
 
+NUMARGS=$#
+ARGS=$*
+
+HELP="build liblegateboost.so and a 'legateboost' Python wheel, and install that wheel
+
+  $0 [<flag> ...]
+
+ where <flag> is any of:
+
+   --editable  install Python wheel in editable mode
+"
+
+function hasArg {
+    (( NUMARGS != 0 )) && (echo " ${ARGS} " | grep -q " $1 ")
+}
```

Comment:
I know some of this looks a little complicated, but it's all standard across most RAPIDS repos. You'll see something similar in `raft`, `cuml`, `cudf`, etc.

example from `raft`: https://github.com/rapidsai/raft/blob/branch-24.10/build.sh

I think consistency with all those projects is useful, as it makes it easier for anyone developing on them to also come help out here, and makes it easier to integrate the various RAPIDS build tools into this repo.

This `build.sh` file will eventually be the one that drives the build for conda packages.